lib/galaxy/jobs/__init__.py

```diff
@@ -786,6 +790,10 @@ def get_parallelism(self):
         return self.tool.parallelism
 
     @property
+    def shell(self):
+        return self.job_destination.shell or getattr(self.app.config, 'default_job_shell', DEFAULT_JOB_SHELL)
```

Comment:
```
    def get( self, key, default ):
        return self.config_dict.get( key, default )
```

I strongly prefer my version. I want the processed value on the config object itself. Also, I like my access pattern assumes less about config in terms of a required interface.